wallet/src/main/scala/org/bitcoins/wallet/internal/TransactionProcessing.scala

```diff
@@ -186,8 +186,21 @@ private[wallet] trait TransactionProcessing extends WalletLogger {
           for {
             outputsBeingSpent <-
               spendingInfoDAO.findOutputsBeingSpent(transaction)
+
+            // unreserved outputs now they are in a block
+            outputsToUse = blockHashOpt match {
```

Comment:
Hmm, so we need to unreserve, and switch to `TxoState.ConfirmedSpent`?

Reply:
We either will in the end set it to `ConfirmedSpent` or `PendingConfirmationsSpent`. However, we set it to `PendingConfirmationsReceived` just to unreserve it and then let `markAsPendingSpent` mark it accordingly